src/service/receiverService.js

```diff
@@ -0,0 +1,113 @@
+import log from "loglevel";
+import { writeFile } from "fs";
+import getMessageCount from "../get/message_count.js";
+import readMessage from "../get/read_message.js";
+import markAsRead from "../put/mark_as_read.js";
+import { checkConnection } from "./helper.js";
+
+export default class receiverService {
+  constructor(config) {
+    this.url = config.url;
+    this.mailboxID = config.mailboxID;
+    this.mailboxPassword = config.mailboxPassword;
+    this.sharedKey = config.sharedKey;
+    this.agent = config.agent;
+
+    let logLevel = process.env.LOG_LEVEL || "DEBUG";
+    log.setLevel(log.levels[logLevel]);
+  }
+
+  async readMessages() {
+    await checkConnection(
+      this.url,
+      this.mailboxID,
+      this.mailboxPassword,
+      this.sharedKey,
+      this.agent
+    );
+
+    // Get the number of messages in mailbox before we add any new ones.
+    log.debug("\nchecking number of messages in mailbox");
+    let inboxCount = await getMessageCount({
+      url: this.url,
+      mailboxID: this.mailboxID,
+      mailboxPassword: this.mailboxPassword,
+      sharedKey: this.sharedKey,
+      agent: this.agent
+    });
+
+    // Loop through the message and read them. so they don't interfere with tests
+    if (inboxCount.data["approx_inbox_count"] > 0) {
+      log.info(
+        "There are " +
+        inboxCount.data["approx_inbox_count"] +
+        " Messages in the mailbox"
+      );
+      log.debug("\nLooping through messages to read their content\n");
+      for (let message of inboxCount.data["messages"]) {
+        let messageResponse = await readMessage({
+          url: this.url,
+          mailboxID: this.mailboxID,
+          mailboxPassword: this.mailboxPassword,
+          sharedKey: this.sharedKey,
+          messageID: message,
+          agent: this.agent
+        })
+        try {
+          if (messageResponse.data === "") {
+            log.warn("WARNING: No data for message " + message);
+          } else if (messageResponse.status === 200) {
+            log.debug("Message ID is: " + message);
```

Comment:
the logic in this condition statement is the same as the one below. Either create a function to move out duplicate code or adjust the conditional statement. When we incorporate sonarqube into this it will flag it up as duplicate code

Reply:
Andrew's implementation, will speak with him and let him know

Reply:
Andrew's code was only a prototype, when you modularised it it should be tidied up.